### PR TITLE
chore: add readme for tool request

### PR DESCRIPTION
**Pull request by sqin2019**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,28 +17,36 @@ to set up this repo properly.**
         request: debug customer issue X".
     -   To request IAM permissions on org/folder/project level, add an
         `iam.yaml` file in the repo root.
-    -   See an example of this file, see example in
-        [example-iam.yaml](https://github.com/abcxyz/aod-template/blob/main/example-iam.yaml).
+    -   To see an example of this file, see example in
+        [example-iam.yaml](example-iam.yaml).
```
Comment (yolocs) [inline README.md]: These two bullet points can be combined. E.g.

```
-   To request IAM permissions on org/folder/project level, add an
    `iam.yaml` file in the repo root. See [example](example-iam.yaml).
```

File: README.md [hunk 2]
```diff
@@ -17,28 +17,36 @@ to set up this repo properly.**
...
     -   (Optional) Use predefined duration labels on the PR to specify the IAM
         permission expiration. Otherwise a 2h default duration will be used.
-    -   (Later) To request on-demand `gcloud` commands, add an `gcloud.yaml`
-        file in the repo root.
+    -   To request on-demand `gcloud` commands, add a `tool.yaml` file
+        in the repo root.
+    -   To see an example of this file, see example in
+        [example-tool.yaml](example-tool.yaml).
```
Comment (yolocs) [inline README.md]: Same comment

File: README.md [hunk 3]
```diff
@@ -17,28 +17,36 @@ to set up this repo properly.**
         request: debug customer issue X".
     -   To request IAM permissions on org/folder/project level, add an
         `iam.yaml` file in the repo root.
-    -   See an example of this file, see example in
-        [example-iam.yaml](https://github.com/abcxyz/aod-template/blob/main/example-iam.yaml).
+    -   To see an example of this file, see example in
+        [example-iam.yaml](example-iam.yaml).
     -   (Optional) Use predefined duration labels on the PR to specify the IAM
```
Comment (yolocs) [inline README.md]: Make this a sub bullet of the IAM file.

File: README.md [hunk 4]
```diff
@@ -17,28 +17,36 @@ to set up this repo properly.**
...
+    -   The result will be posted as PR comments.
 
-4.  The AOD request PR cannot and should not be merged. Close the PR after you
-    no longer need the access. The PR will automatically be closed after X
-    hours.
+4.  The AOD request PR cannot and should not be merged
+
+    -   Close the IAM request PR after you no longer need the access.
+    -   Close the tool request PR when you want to execute the cleanup commands
+        (`cleanup` commands in your `tool.yaml` file).
```
Comment (yolocs) [inline README.md]: IAM and tool requests are not exclusive, aren't they? I'd simply say close you the PR when you're done.

And then mention that a workflow will be triggered to do clean up

**Review by yolocs — CHANGES_REQUESTED** [review]

File: example-tool.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Tool request example file.
+
+# Currently we only support gcloud cli, both `do` and `cleanup` commands need to
+# be without the cli name.
+
+# See below example:
+
+do:
```
Comment (yolocs) [inline example-tool.yaml:6]: Don't we need to specify the tool?

File: README.md [hunk 6]
```diff
@@ -97,4 +102,6 @@ The admin of your GCP project/folder/org to complete the steps below.
         -   Disallow force pushes
         -   Disallow deletions
 
-## Adding code other than AOD (TODO #9)
+## Adding code other than AOD
+
+It should be the same as you adding code in your other repositories.
```
Comment (yolocs) [inline README.md]: If the branch is locked, how to add code?

**Review by yolocs — DISMISSED** [review]

File: example-tool.yaml (hunk 5 above)
Comment (sqin2019) [reply example-tool.yaml:6]: default is gcloud.

File: README.md (hunk 6 above)
Comment (sqin2019) [reply README.md]: Adding more details.

File: example-tool.yaml (hunk 5 above)
Comment (yolocs) [reply example-tool.yaml:6]: I think it's better to have the explicit value here. And comment that the default value is gcloud

File: README.md (hunk 6 above)
Comment (yolocs) [reply README.md]: But if branch is locked, passing all checks still won't allow you to merge.

Comment (sqin2019) [reply README.md]: Ah..forgot the branch is locked. It needs the PR author to ask for bypass branch protections permission. Wouldn't it bypass all protections including require PR, checks and approvals?

Comment (sqin2019) [reply README.md]: Maybe we don't need to provide too much detail, so the recommended way is to use [restrict-who-can-push-to-matching-branches](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches#restrict-who-can-push-to-matching-branches) setting to allow a certain group to push to main but still require PR, checks and approval. PR authors can request to be added to that group temporary for merging a code change?

File: example-tool.yaml (hunk 5 above)
Comment (sqin2019) [reply example-tool.yaml:6]: Sure.

File: README.md [hunk 7]
```diff
@@ -16,29 +16,32 @@ to set up this repo properly.**
...
-        permissions or run the on-demand `gcloud` commands
-    -   The result will be posted as PR comments
+        permissions or run the on-demand `gcloud` commands (`do` commands in
+        your `tool.yaml` file).
+    -   The result will be posted as PR comments.
 
-4.  The AOD request PR cannot and should not be merged. Close the PR after you
-    no longer need the access. The PR will automatically be closed after X
-    hours.
+4.  The AOD request PR cannot and should not be merged
```
Comment (capri-xiyue) [inline README.md]: nits: use Bold Uppercase Fonts for `cannot` and `should not`

**Review by capri-xiyue — DISMISSED** [review]

File: README.md [hunk 8]
```diff
@@ -97,4 +102,10 @@ The admin of your GCP project/folder/org to complete the steps below.
...
         -   Disallow deletions
 
-## Adding code other than AOD (TODO #9)
+## Adding code other than AOD
+
+You need to have:
+
+-   A PR ready to merge: open a PR with signed commits -> pass status checks if
+    any -> get required approvals.
+-   Permissions to bypass the `lock branch` branch protection rule.
```
Comment (yolocs) [inline README.md]:
```suggestion
Rarely but the repo admins might need to check in changes to the repo that are not AOD request.
To do that, the admin would need to send PRs as usual with "lock branch" temporarily disabled.
This activity should be coordinated carefully to make sure AOD request PRs won't be merged accidentally.
```

**Review by yolocs — DISMISSED** [review]

**Review by yolocs — APPROVED** [review]

**Review by capri-xiyue — APPROVED** [review]

**Pull request merged**